angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit, HostBinding, Input, Output, EventEmitter, ViewContainerRef, TemplateRef, ViewChild, ElementRef } from '@angular/core';
+import { TimePickerService, MinuteIntervalType } from './time-picker.service';
+import {
+  Overlay,
+  OverlayConfig,
+  OverlayRef
+} from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+export declare type HorizontalConnectionPos = 'start' | 'center' | 'end';
+export declare type VerticalConnectionPos = 'top' | 'center' | 'bottom';
+
+@Component({
+  selector: 'cui-time-picker',
+  template: `
+    <div class='cui-input-group  read-only' #connectToInput>
+      <input class="cui-input read-only dirty" 
+      tabindex="0" 
+      type="text" 
+      name="cui-timepicker__input-407" 
+      readonly=""
+      (focus) = 'onFocus()'
+      [value]="value">
+    </div>
+    <ng-template #timePickerCon>
+      <cui-time-picker-dropdown>
+        <cui-time-selector unit='h'></cui-time-selector>:
+        <cui-time-selector unit='m'></cui-time-selector>
+        <cui-time-selector unit='pre' *ngIf='!militaryTime'></cui-time-selector>
+      </cui-time-picker-dropdown>
+    </ng-template>
+  `,
+  styles: [],
+  providers:[TimePickerService]
+})
+export class TimePickerComponent implements OnInit {
```

Comment:
Needs ngModel/formControl support, so that you can use this along with Angular Forms